streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -240,8 +240,8 @@
                     STATE_DIR_DOC)
             .define(REPLICATION_FACTOR_CONFIG,
                     Type.INT,
-                    1,
-                    Importance.MEDIUM,
+                    3,
```

Comment:
The code actually uses min(#brokers, REPLICATION_FACTOR), and prints a warning, but it still runs with, say, 1 broker.